app/uk/gov/hmrc/transitmovementsauditing/models/AuditType.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsauditing.models
+
+sealed trait AuditType
+
+object AuditType extends Enumeration {
+
+  val AmendmentAcceptance, ArrivalNotification, InvalidationDecision, DeclarationAmendment, DeclarationInvalidationRequest, DeclarationData, Discrepancies,
```

Comment:
I'm torn about this, because I understand the horridness of doing this via sealed traits/classes, but we don't do things this way elsewhere. We've tended to create a sealed trait and inherit from that, such as:

```scala
sealed abstract class ObjectType(val name: String) extends Product with Serializable

object AuditType {
    
    val values: Seq[AuditType] = Seq(AmendmentAcceptance, ...)

    case object AmendmentAcceptance extends AuditType("AmendmentAcceptance")
    case object ArrivalNotification           extends AuditType("ArrivalNotification")
    ...

    def fromName(name: String): Option[AuditType] = values.find(_.name == name)
}
```

I think we should probably stick with the route we've taken before, especially as there are other drawbacks to enumerations in Scala (the main one being that all enumerations compile to the same type erasure, and so pattern matching case statements can't be checked for completeness).

Reply:
Ok, will change to reflect this.